app/controllers/PartTimePeriodsController.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import java.time.LocalDate
+
+import cats.data.Validated.{Invalid, Valid}
+import controllers.actions._
+import forms.{AdditionalPaymentPeriodsFormProvider, PartTimePeriodsFormProvider}
+import handlers.{ErrorHandler, FurloughCalculationHandler}
+import javax.inject.Inject
+import models.{PartTimePeriods, Periods, UserAnswers}
+import navigation.Navigator
+import pages.{AdditionalPaymentPeriodsPage, PartTimePeriodsPage, PayDatePage}
+import play.Logger
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import repositories.SessionRepository
+import services.{FurloughPeriodExtractor, PeriodHelper}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.{AdditionalPaymentPeriodsView, PartTimePeriodsView}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PartTimePeriodsController @Inject()(
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  override val navigator: Navigator,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: PartTimePeriodsFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: PartTimePeriodsView
+)(implicit ec: ExecutionContext, errorHandler: ErrorHandler)
+    extends BaseController with PeriodHelper with FurloughPeriodExtractor {
+
+  val form: Form[List[LocalDate]] = formProvider()
+
+  def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    request.userAnswers.getList(PayDatePage) match {
+      case _ :: date :: Nil =>
+        saveAndRedirect(request.userAnswers, List(date))
```

Comment:
is this method `saveAndRedirect` causing a double redirect to the same page in this case ?

Reply:
I don't think so, this is checking if there is only 1 pay period, 1 pay period is made up from 2 pay period end dates which is what we are retrieving from userAnswers. hence the _ :: date ... the saveAndRedirect just does the nasty stuff of generating the pay periods and stores in mongo as a Full/Partial period